### ISSUE TITLE
Question about the global std

### ISSUE THREAD

**Issue by Shredded-Pork** [yifan123/flow_grpo#52]
I notice that you compute the std with all data in a batch not per prompt. Why do global std？I use per prompt and obtain the better performance. The purple line is consistent with the result in your paper in PickScore and the blue line is with per prompt std setting.

<img width="1249" height="698" alt="Image" src="https://github.com/user-attachments/assets/8231f532-f706-4d6b-b9a7-f771a5b94f9d" />

**Comment by Shredded-Pork** [comment]
I also check the code of dance-grpo, and it uses the group std not global std. The training steps is about 300-500. So I think the correct setting is using group std and train about 500-1k steps? In your setting, there are  too many steps😂.

**Comment by yifan123** [comment]
Thank you very much for your experimental results.

The global standard deviation trick is inspired by an LLM paper: *"REINFORCE++: An Efficient RLHF Algorithm with Robustness to Both Prompt and Reward Models* [(https://arxiv.org/pdf/2501.03262)](https://arxiv.org/pdf/2501.03262). This paper says that "employs a global batch mean reward as a baseline to prevent overfitting and instability". I didn’t perform a strict ablation to test its effectiveness for flow models and simply adopted it as a practical heuristic.

I really appreciate your findings — I’ll further investigate whether this trick holds across a broader range of tasks and will credit you in the README. Also, the latest version of FlowGRPO now supports Flux. I’ve found that Flux is easier to train than SD3 and requires fewer steps.

**Comment by yifan123** [comment]
May I kindly ask what beta values you used for your two curves?

**Comment by Shredded-Pork** [comment]
config.train.beta = 0.001 
It notices that I didn't change the std_dev_t, so the beta is smaller than your config.

**Comment by Shredded-Pork** [comment]
REINFORCE++: An Efficient RLHF Algorithm with Robustness to Both Prompt and Reward Models [(https://arxiv.org/pdf/2501.03262)](https://arxiv.org/pdf/2501.03262). This paper says that "employs a global batch mean reward as a baseline to prevent overfitting and instability. In your code, you use the mean in group but std in batch, is there any mistake.

**Comment by Shredded-Pork** [comment]
As for KL loss, I have the same observation as you. The kl of local std is much larger than global std. As shown in the above issue, I think if your use global std, which means that the adv. is dived by a larger value than its original std. This would make the weight of policy loss to be smaller, so I think you can adjust the coe of kl to make the same result.

**Comment by yifan123** [comment]
Yes, the results from my own training match your reward curve and KL curve exactly.
